ex_extra_diagnostics compile surport

y123456yz opened the pull request:
ex_extra_diagnostics compile surport

y123456yz commented:
![image](https://github.com/wiredtiger/wiredtiger/assets/14356223/e63f7fe5-23fb-4e51-8102-c06e1277b8d8)

y123456yz commented:
https://jira.mongodb.org/browse/WT-11909

keitharnoldsmith commented:
@y123456yz, thank you for the submission. It looks like your patch is missing the `ex_extra_diagnostics.c` file. Can you please update the PR?

y123456yz commented:
> @y123456yz, thank you for the submission. It looks like your patch is missing the `ex_extra_diagnostics.c` file. Can you please update the PR?

hi, @keitharnoldsmith 
develop branch Already have the `ex_extra_diagnostics.c` file, The only thing missing here is the compiled code. thanks very much. 
https://github.com/wiredtiger/wiredtiger/blob/develop/examples/c/ex_extra_diagnostics.c

ajmorton commented:
Yep, `ex_extra_diagnostics.c` already existed, we added it for the code snippets in the `Configuring runtime diagnostics` section of the [debugging page](https://source.wiredtiger.com/develop/debugging.html), but we should also add it as a runnable example.

The existing code fails when run in diagnostic mode as this config item only makes sense in release mode. I've updated `ex_extra_diagnostics.c` to handle this case

y123456yz commented:
> Yep, `ex_extra_diagnostics.c` already existed. We added it for the code snippets in the `Configuring runtime diagnostics` section of the [debugging page](https://source.wiredtiger.com/develop/debugging.html), but we should also add it as a runnable example.
> 
> The existing code fails when run in diagnostic mode as this config item only makes sense in release mode. I've updated `ex_extra_diagnostics.c` to handle this case

got it ,thanks.

ajmorton reviewed (approved):
LGTM

ajmorton commented:
evergreen merge
